src/profile-logic/process-profile.js

```diff
@@ -1717,19 +1718,53 @@ export function processGeckoProfile(geckoProfile: GeckoProfile): Profile {
   return result;
 }
 
+function _serializeSamples({ time, ...restOfSamples }): any {
+  let lastTime;
+  return {
+    timeDeltas: time.map((t, i) => {
+      const timeDelta = i === 0 ? t : t - lastTime;
+      lastTime = t;
+      return timeDelta;
+    }),
+    ...restOfSamples,
+  };
+}
+
+function _unserializeSamples({ timeDeltas, time, ...restOfSamples }): any {
+  let lastTime;
+  return {
+    time:
+      time ||
+      timeDeltas.map((t, i) => {
+        return (lastTime = i === 0 ? t : lastTime + t);
+      }),
+    ...restOfSamples,
+  };
+}
```

Comment:
In both `_serializeSamples` and `_unserializeSamples` you can initialize lastTime to zero and remove the `i === 0` check.